python/lsst/validate/drp/calcsrd/amx.py

```diff
@@ -198,26 +198,83 @@ def magInRange(cat):
                                      (dist < annulusRadians[1]))
         for obj2 in objectsInAnnulus:
-            distances = matchVisitComputeDistance(
-                visit[obj1], ra[obj1], dec[obj1],
-                visit[obj2], ra[obj2], dec[obj2])
-            if not distances:
+            rmsDist = calcRmsDistanceForOneObject(visit, ra, dec, obj1, obj2)
+            if rmsDist is None:
                 if verbose:
-                    print("No matching visits found for objs: %d and %d" %
-                          (obj1, obj2))
+                    print("No matching visits found for objs: %d and %d" % (obj1, obj2))
                 continue
 
-            finiteEntries, = np.where(np.isfinite(distances))
-            # Need at least 2 distances to get a nonzero stdev
-            if len(finiteEntries) > 1:
-                rmsDist = np.std(np.array(distances)[finiteEntries])
-                rmsDistances.append(rmsDist)
+            rmsDistances.append(rmsDist)
 
     # return quantity
     rmsDistances = np.array(rmsDistances) * u.radian
     return rmsDistances
 
 
+def calcRmsDistanceForOneObject(visit1, ra1, dec1, visit2, ra2, dec2,
+                                verbose=False):
+    """Calculate the RMS of the distance between two objects over visits.
+
+    Parameters
+    ----------
+    visit1 : visit IDs of object 1
+    ra1 : RAs of object 1
+    dec1 : DECs of object 1
+    visit2 : visit IDs of object 2
+    ra2 : RAs of object 2
+    dec2 : DECs of object 2
+    verbose : bool, optional
+        Output additional information on the analysis steps.
+
+    visit1 and visit2 should share at least two common visits to return a result.
+    They don't have to be in order, they can be different sets.  They should
+    just overlap by at least two common visits.  If the overlap is less, then
+    this will return None.
+
+    Returns
+    -------
+    rmsDistances : RMS of angular separation in radians.
+        return None if no more than 1 distance.
+
+    Examples
+    --------
+    2e-5 rad (4") change in RA offset between stars across visits
+        (Dec 0 so cos(dec)=1).
+    So each is 1e-5 off from average.
+    So the RMS should be sqrt(2) * 10**(-5)
+    >>> visit1, ra1, dec1 = [1, 2], [10.12344, 10.12345], [0, 0]
+    >>> visit2, ra2, dec2 = [1, 2], [20.00000, 19.99999], [0, 0]
+    >>> r = calcRmsDistanceForOneObject(visit1, ra1, dec1, visit2, ra2, dec2)
+    >>> print(r)
+    1.4142135622881543e-05
+
+    Moved together so should be RMS=0.
+    Same as above, but switched visit IDs for object 2.
+    Testing that visit ID is tracked properly
+    >>> visit1, ra1, dec1 = [1, 2], [10.12344, 10.12345], [0, 0]
+    >>> visit2, ra2, dec2 = [2, 1], [20.00000, 19.99999], [0, 0]
+    >>> r = calcRmsDistanceForOneObject(visit1, ra1, dec1, visit2, ra2, dec2)
+    >>> print(r)
+    0.0
+
+    Should return None
+    >>> visit1, ra1, dec1 = [1], [10.12344], [0, 0]
+    >>> visit2, ra2, dec2 = [1], [20.00000], [0, 0]
```

Comment:
As you likely suspect, this was a mistake.  I didn't mean to test this. 

I'm not surprised that it works because `matchVisitComputeDistance` iterates over visits and indexes `ra` and `dec` by that index, so if `dec` is longer then it never accesses those extra entries.